cabby/data/extract.py

```diff
@@ -14,140 +14,197 @@
+    qid(Text): The qid of the Wikidata to extract items from.
+  Returns:
+    The Wikipedia (text,title) and Wikidata (location) data found.
+  '''
 
-    # Get Wikidata items by region.
-    wikidata_results = wdq.get_geofenced_wikidata_items_by_qid(qid)
-    wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
-                      for result in wikidata_results]
+  # Get Wikidata items by region.
+  wikidata_results = wdq.get_geofenced_wikidata_items_by_qid(qid)
+  wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
+            for result in wikidata_results]
 
-    return get_wikigeo_data(wikidata_items)
+  return get_wikigeo_data(wikidata_items)
 
 
 def get_data_by_region(region: Text) -> Sequence:
-    '''Get data from Wikipedia and Wikidata by region" 
-    Arguments:
-        region(Text): The region to extract items from.
-    Returns:
-        The Wikipedia (text,title) and Wikidata (location) data found.
-    '''
-
-    # Get Wikidata items by region.
-    wikidata_results = wdq.get_geofenced_wikidata_items(region)
-    wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
-                      for result in wikidata_results]
-
-    return get_wikigeo_data(wikidata_items)
-
-
-def split_dataset( 
+  '''Get data from Wikipedia and Wikidata by region" 
+  Arguments:
+    region(Text): The region to extract items from.
+  Returns:
+    The Wikipedia (text,title) and Wikidata (location) data found.
+  '''
+
+  # Get Wikidata items by region.
+  wikidata_results = wdq.get_geofenced_wikidata_items(region)
+  wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
+            for result in wikidata_results]
+
+  return get_wikigeo_data(wikidata_items)
+
+
+def get_data_by_region_expansive(region: Text, path_osm: Text) -> Sequence:
+  '''Get data from Wikipedia and Wikidata by region and add sampes created from 
+  Wikidata tags." 
+  Arguments:
+    region(Text): The region to extract items from.
+  Returns:
+    The Wikipedia (text,title) and Wikidata (location) data found.
+  '''
+
+
+  # Get Wikidata items by region.
+  wikidata_results = wdq.get_geofenced_wikidata_items(region)
+  wikidata_items = [wdi.WikidataEntity.from_sparql_result(result)
+            for result in wikidata_results]
+
+  samples = get_wikigeo_data(wikidata_items)
+
+  logging.info(
+    'Created {} samples from Wikipedia pages and Wikidata \
+    (locations only).'.format(len(samples)))
+
+  # Add samples from Wikidata only.
+  wikidata_tags = wdq.get_geofenced_info_wikidata_items(region)
+  for item in wikidata_tags:
+    info_item = wdqi.WikidataEntity.from_sparql_result_info(item)
+    sample = wikigeo.WikigeoEntity.from_wikidata(info_item).sample
+    samples.append(sample)
+
+  logging.info(
+    'Created {} samples with Wikidata additional data.'.format(len(samples)))
+
+  # Add sample from OSM only.
+  poi = map_structure.load_poi(path_osm)
+  poi = poi[poi['s2cellids'].str.len() <= 10]  # Remove large entities.
```

Comment:
may be define a constant for this?
Also lets count these examples. Are there too many? may be affected by how the OSM entities are named. or rather how certain type of wiki entities are named.